### PR TITLE
Implement preview API to retrieve licenses

### PR DESCRIPTION
**Pull request by Haacked**
This implements the miscellaneous API endpoints for retrieving license info. This API is a PREVIEW api.

https://developer.github.com/v3/licenses/

### CODE REVIEW

File: Octokit.Reactive/Clients/ObservableMiscellaneousClient.cs [hunk 1]
```diff
@@ -24,5 +24,27 @@ public IObservable<string> RenderRawMarkdown(string markdown)
...
+
+        /// <summary>
+        /// Returns a list of the licenses shown in the license picker on GitHub.com. This is not a comprehensive
+        /// list of all possible OSS licenses.
+        /// </summary>
+        /// <remarks>This is a PREVIEW API! Use it at your own risk.</remarks>
+        /// <returns>A list of licenses available on the site</returns>
+        public IObservable<LicenseMetadata> GetLicenses()
+        {
+            return _client.GetLicenses().ToObservable().SelectMany(l => l);
```
Comment (shiftkey) [inline Octokit.Reactive/Clients/ObservableMiscellaneousClient.cs]: :cry:

**Comment by shiftkey** [comment]
Code looks good. Just waiting on the green build and the merge conflict, otherwise :thumbsup:

File: Octokit.Reactive/Clients/ObservableMiscellaneousClient.cs (hunk 1 above)
Comment (Haacked) [inline Octokit.Reactive/Clients/ObservableMiscellaneousClient.cs]: Tears? What for?

**Comment by Haacked** [comment]
I need to rebase this real quick.

File: Octokit.Reactive/Clients/ObservableMiscellaneousClient.cs (hunk 1 above)
Comment (shiftkey) [inline Octokit.Reactive/Clients/ObservableMiscellaneousClient.cs]: Just general sadness at the need to `SelectMany`. It's fiiiiine.

File: Octokit/Models/Response/License.cs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+        /// <summary>
+        /// Set of codes for what is forbidden under the terms of the license. For example, "no-liability"
+        /// </summary>
+        public IReadOnlyCollection<string> Forbidden { get; protected set; }
+
+        /// <summary>
+        /// The text of the license
+        /// </summary>
+        public string Body { get; set; }
```
Comment (shiftkey) [inline Octokit/Models/Response/License.cs]:
```
   Octokit.Tests.Conventions.ModelTests.ResponseModelsHaveGetterOnlyProperties(modelType: typeof(Octokit.License)) [FAIL]
      Assert.True() Failure
      Stack Trace:
         Octokit.Tests.Conventions\ModelTests.cs(31,0): at Octokit.Tests.Conventions.ModelTests.ResponseModelsHaveGetterOnlyProperties(Type modelType)
Finished: Octokit.Tests.Conventions.DLL
```

Comment (khellang) [inline Octokit/Models/Response/License.cs]: Gotta love those convention tests :wink:

Comment (shiftkey) [inline Octokit/Models/Response/License.cs]: :star2:

**Comment by Haacked** [comment]
:station:

File: Octokit/Clients/MiscellaneousClient.cs [hunk 3]
```diff
@@ -82,7 +82,38 @@ public MiscellaneousClient(IConnection connection)
             var endpoint = new Uri("gitignore/templates/" + Uri.EscapeUriString(templateName), UriKind.Relative);
 
             var response = await _connection.Get<GitIgnoreTemplate>(endpoint, null, null)
-                                  .ConfigureAwait(false);
+                .ConfigureAwait(false);
```
Comment (shiftkey) [inline Octokit/Clients/MiscellaneousClient.cs]: :lipstick: indenting

Comment (Haacked) [inline Octokit/Clients/MiscellaneousClient.cs]: Yeah, this is now correct, no? Before it wasn't.

Comment (shiftkey) [inline Octokit/Clients/MiscellaneousClient.cs]: I'm indifferent on which way is correct, but either this one is right or the new code below is right.

Comment (Haacked) [inline Octokit/Clients/MiscellaneousClient.cs]: Ha! I let R# sort it out.

Comment (shiftkey) [inline Octokit/Clients/MiscellaneousClient.cs]: :sparkles:

**Comment by shiftkey** [comment]
And by YOLO I mean "I'm gonna look into why this broke SourceLink"

**Pull request merged**